sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala

```diff
@@ -51,32 +52,17 @@ private[lf] final class CommandPreprocessor(
 
   @throws[Error.Preprocessing.Error]
   def unsafePreprocessDisclosedContract(
-      disc: command.DisclosedContract
+      disc: FatContractInstance
   ): speedy.DisclosedContract = {
-    val tmpl = handleLookup(pkgInterface.lookupTemplate(disc.templateId))
-    (tmpl.key, disc.keyHash) match {
-      case (Some(_), None) =>
-        throw Error.Preprocessing.MissingDisclosedContractKeyHash(
-          disc.contractId,
-          disc.templateId,
-        )
-      case (None, Some(hash)) =>
-        throw Error.Preprocessing.UnexpectedDisclosedContractKeyHash(
-          disc.contractId,
-          disc.templateId,
-          hash,
-        )
-      case _ =>
-    }
     // TODO: https://github.com/digital-asset/daml/issues/17082
     //  for now we need the package of the disclosed contract
-    val arg = translateUpgradableArg(Ast.TTyCon(disc.templateId), disc.argument, strict = true)
+    val arg = translateUpgradableArg(Ast.TTyCon(disc.templateId), disc.createArg, strict = true)
```

Comment:
This uses the original template. Now we have a package name can we resolve to a template that is in the dependency tree?